### PR TITLE
[Foxy] backport pluginlib enable plugin testing

### PR DESCRIPTION
**Pull request by sloretz**
This cherry-pick backports #201 to ROS Foxy

### CODE REVIEW

**Comment by sloretz** [comment]
Should cherry-pick #203 before backporting

**Comment by sloretz** [comment]
Cherry-picked 8b5a1a2 - ready for review again.

File: pluginlib/cmake/pluginlib_enable_plugin_testing.cmake [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    else()
+      configure_file("${input_file_abs}" "${output_file}" COPYONLY)
+    endif()
+  endforeach()
+
+  # Copy library targets at build time; but make directory at configure time
+  # because `cmake -E copy` does not make directories
+  file(MAKE_DIRECTORY "${prefix}/lib")
+  foreach(plugin_library "${ARG_PLUGIN_LIBRARIES}")
+    set(output_library "${prefix}/lib/$<TARGET_FILE:${plugin_library}>")
```
Comment (hidmic) [inline pluginlib/cmake/pluginlib_enable_plugin_testing.cmake]: @sloretz unused?

Comment (sloretz) [reply pluginlib/cmake/pluginlib_enable_plugin_testing.cmake]: Oops, yup. Opened PR to remove it on `ros2` branch; will include it into the backport when that's merged.

https://github.com/ros/pluginlib/pull/211

**Comment by sloretz** [comment]
Cherry-picked #211 onto this branch too

**Comment by jacobperron** [comment]
Friendly ping. What's the status of this PR?

**Comment by sloretz** [comment]
I definitely forgot about it. It looks like feedback has been addressed, so it needs a review and CI

**Review by jacobperron — APPROVED** [review]
LGTM

**Comment by sloretz** [comment]
CI (Foxy repos, build: `--packages-up-to pluginlib` test: `--packages-select pluginlib`)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15729/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10411/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13394/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15973/)

**Pull request merged**